static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
-    undefined
-  );
+    // Show system-level alerts
+    config.messages.forEach(msg =>
+      AlertActions.addAlert({message: msg.message, type: msg.level, neverExpire: true})
+    );
 
-  handleConfigStoreChange(config) {
-    const newState = {} as State;
-    if (config.needsUpgrade !== undefined) {
-      newState.needsUpgrade = config.needsUpgrade;
+    // The app is running in deploy preview mode
+    if (DEPLOY_PREVIEW_CONFIG) {
+      displayDeployPreviewAlert();
     }
 
-    if (config.user !== undefined) {
-      newState.user = config.user;
+    // The app is running in local SPA mode
+    if (!DEPLOY_PREVIEW_CONFIG && EXPERIMENTAL_SPA) {
+      displayExperimentalSpaAlert();
     }
 
-    if (Object.keys(newState).length > 0) {
-      this.setState(newState);
+    // Set the user for analytics
+    if (config.user) {
+      HookStore.get('analytics:init-user').map(cb => cb(config.user));
     }
-  }
 
-  @keydown('meta+shift+p', 'meta+k', 'ctrl+shift+p', 'ctrl+k')
-  openCommandPalette(e) {
-    openCommandPalette();
-    e.preventDefault();
-    e.stopPropagation();
-  }
-
-  @keydown('meta+shift+l', 'ctrl+shift+l')
-  toggleDarkMode() {
-    ConfigStore.set('theme', ConfigStore.get('theme') === 'light' ? 'dark' : 'light');
-  }
+    initApiClientErrorHandling();
+    fetchGuides();
 
-  onConfigured = () => this.setState({needsUpgrade: false});
+    // When the app is unloaded clear the organizationst list
+    return () => OrganizationsStore.load([]);
+  }, []);
 
-  // this is somewhat hackish
-  handleNewsletterConsent = () =>
-    this.setState({
-      newsletterConsentPrompt: false,
-    });
+  function clearUpgrade() {
+    ConfigStore.set('needsUpgrade', false);
+  }
 
-  handleGlobalModalClose = () => {
-    if (typeof this.mainContainerRef.current?.focus === 'function') {
-      // Focus the main container to get hotkeys to keep working after modal closes
-      this.mainContainerRef.current.focus();
-    }
-  };
+  function clearNewsletterConsent() {
+    const flags = {...config.user.flags, newsletter_consent_prompt: false};
+    ConfigStore.set('user', {...config.user, flags});
+  }
 
-  renderBody() {
-    const {needsUpgrade, newsletterConsentPrompt} = this.state;
+  const needsUpgrade = config.user?.isSuperuser && config.needsUpgrade;
+  const newsletterConsentPrompt = config.user?.flags?.newsletter_consent_prompt;
 
+  function renderBody() {
     if (needsUpgrade) {
       const InstallWizard = lazy(() => import('app/views/admin/installWizard'));
 
       return (
         <Suspense fallback={null}>
-          <InstallWizard onConfigured={this.onConfigured} />;
+          <InstallWizard onConfigured={clearUpgrade} />;
         </Suspense>
       );
     }
 
     if (newsletterConsentPrompt) {
-      return <NewsletterConsent onSubmitSuccess={this.handleNewsletterConsent} />;
-    }
-
-    return this.props.children;
-  }
+      const NewsletterConsent = lazy(() => import('app/views/newsletterConsent'));
```

Comment:
I converted NewsletterConsent to a lazy-load. I will need to actually verify that this works.